contracts/voting-verifier/src/events.rs

```diff
@@ -77,6 +95,24 @@ impl TryFrom<Vec<Message>> for EvmMessages {
     }
 }
 
+#[cw_serde]
+pub struct WorkerSetConfirmation {
+    pub tx_id: String,
+    pub log_index: u64,
+    pub operators: Operators,
+}
+
+impl WorkerSetConfirmation {
+    pub fn new(message_id: MessageID, operators: Operators) -> Result<Self, ContractError> {
+        let (tx_id, log_index) = parse_message_id(message_id.to_string())?;
```

Comment:
Why does `parse_message_id` take a `String` instead of `&MessageID` or even `AsRef<MessageID>`?